### DISCUSSION TITLE
Citation style in typst seems to be inconsistent with documentation

### DISCUSSION THREAD

**Discussion by aghaynes** [quarto-dev/quarto-cli#9761] [Q&A] [answered]
### Description

The documentation on the quarto website about citations suggests that changing the citation style is just a case of specifying the style in the csl argument.

For typst, this doesnt seem to work. It uses the bibliographystyle argument instead (unless you set citeproc to true as a suboption to the format). Switching between the different styles named in `csl` in the example below doesnt seem to do much. I'm not sure whether that's a bug or a feature or...?

````qmd
---
title: "Untitled"
format: typst
bibliography: references.bib
# csl: begell-house-chicago-author-date.csl
csl: nature.csl
# bibliographystyle: nature.csl
# csl: apa.csl
---


@r2023 says x 

bla bla [@r2023]

bla bla [see @r2023]

la la la ([@r2023; @r2024])

The R manual ([-@r2023]) says X
```` 

references.bib:
````
@Manual{r2023,
    title = {R: A Language and Environment for Statistical Computing},
    author = {{R Core Team}},
    organization = {R Foundation for Statistical Computing},
    address = {Vienna, Austria},
    year = {2023},
    url = {https://www.R-project.org/},
  }
@Manual{r2024,
    title = {R: A Language and Environment for Statistical Computing},
    author = {{R Core Team}},
    organization = {R Foundation for Statistical Computing},
    address = {Vienna, Austria},
    year = {2024},
    url = {https://www.R-project.org/},
  }
````

**Comment by mcanouil** [comment, marked as answer]
> The documentation on the quarto website about citations suggests that changing the citation style is just a case of specifying the style in the csl argument.

Where did you see that?

The behaviour is documented: <https://quarto.org/docs/output-formats/typst.html#bibliography>.

**Reply by aghaynes** [reply]
Ah, OK... Maybe a note could be made on the main citations page that typst is different? https://quarto.org/docs/authoring/footnotes-and-citations.html

**Reply by cderv** [reply]
For PDF with LaTeX, we document native citations: https://quarto.org/docs/output-formats/pdf-basics.html#citations
but we also add it to main doc: https://quarto.org/docs/output-formats/pdf-basics.html#citations

It could make sense to do the same for typst... 🤔 

I'll keep the idea. Thanks ! 

By the way, I guess you saw that if you want to use CSL for Pandoc citeproc, you need to set `citeproc: true` in Typst to opt-in. By default, the native citation processing for typst will be used.

**Reply by aghaynes** [reply]
Thanks @cderv, yes, I found that.

**Reply by cderv** [reply]
https://github.com/quarto-dev/quarto-cli/issues/9762
